### PR TITLE
One to clone them all

### PR DESCRIPTION
**Pull request by amuniz**
Simple script to clone all plugins which this one aggregates.

Usage: `./clone-all` or `./clone-all amuniz` (which clones `git@github.com:amuniz/[plugins_set]`

@reviewbybees esp. @jglick

### CODE REVIEW

File: clone-all [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+username='jenkinsci'
+if [ -z "$1" ]; then
+  echo "Using jenkinsci/* for clones";
+else
+  echo "Using '$1' for clones";
+  username=$1
+fi
+
+git clone git@github.com:$username/workflow-step-api-plugin.git ../workflow-step-api-plugin
```
Comment (jglick) [inline clone-all]: More maintainable use Perl (or Ruby, or Bash + `sed`, or whatever) to grep `demo/plugins.txt` (and, at least until #1 is merged, `demo/snapshot-only-plugins.txt`) for `workflow-*` or `pipeline-*` plugins.

The exception is `pipeline-stage-view`, which together with `pipeline-rest-api` lives in `pipeline-stage-view-plugin`. Not currently a dependency here, but probably will be.

Also you forget `workflow-multibranch`, it seems.

**Comment by jglick** [comment]
You know about [`clone.groovy`](https://github.com/jenkinsci/backend-merge-all-repo/blob/66d6e33c1a0a3abdbd6573abb0077a9132656455/clone.groovy), right?

File: clone-all (hunk 1 above)
Comment (amuniz) [inline clone-all]: Ok. Will try that.

**Comment by amuniz** [comment]
> You know about clone.groovy, right?

Yes, I know, you told me in London I think, but I don't want to clone the whole organization.

**Comment by batmat** [comment]
Wouldn't it be possibly simpler and more standard to use git submodules?

This way, you'd do `git clone --recursive https://github.com/jenkinsci/workflow-aggregator-plugin` and get the same result as this script IIUC?

**Comment by amuniz** [comment]
> This way, you'd do git clone --recursive https://github.com/jenkinsci/workflow-aggregator-plugin and get the same result as this script IIUC?

Not really, as I would want to clone my forks.

**Comment by batmat** [comment]
Ack. I think this may still be possible with submodules I guess, but possibly also having a custom script may be more flexible in the long term.

**Comment by jtnord** [comment]
Can we have a file extension please, that way people who have funky tab auto completion can get auto complete for it

**Comment by jtnord** [comment]
>> This way, you'd do git clone --recursive https://github.com/jenkinsci/workflow-aggregator-plugin and get the same result as this script IIUC?
> Not really, as I would want to clone my forks.

Well you still normally want origin to make sure your code is up to date, so have to then add the remote to your modules.

`git submodule foreach` is your friend.

you can run lots of whatever you want with submodules.
e.g. 

```
git submodule foreach 'git remote add git@github.com:$USER/$name'
# $name is the name of the submodule - so set this to the same as the remote repo name
git submodule foreach 'git fetch'
git submodule foreach 'git reset --hard origin/master'
git submodule foreach 'git checkout -b JENKINS-XYZ' 
```

Anyway - this is just a comment not anything on the PR and submodules do have their own behaviour :)

**Comment by jglick** [comment]
:bee:

**Pull request merged**